ncclient/transport/third_party/junos/parser.py

```diff
@@ -0,0 +1,269 @@
+# Copyright 2018 Nitin Kumar
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+
+from threading import Lock
+
+import difflib
+
+from xml.sax.handler import ContentHandler
+from lxml import etree
+from lxml.builder import E
+from xml.sax._exceptions import SAXParseException
+from xml.sax import make_parser
+
+from ncclient.transport.parser import DefaultXMLParser
+from ncclient.operations import rpc
+from ncclient.transport.parser import SAXFilterXMLNotFoundError
+from ncclient.operations.errors import OperationError
+
+import logging
+logger = logging.getLogger("ncclient.transport.third_party.junos.parser")
+
+# v1.0: RFC 4742
+MSG_DELIM = "]]>]]>"
```

Comment:
this is common across both parsers; should it be in a separate common module?

Reply:
Fixed via https://github.com/ncclient/ncclient/pull/324/commits/9fcad710757bf774bd672d8253e1c42b6083eda1